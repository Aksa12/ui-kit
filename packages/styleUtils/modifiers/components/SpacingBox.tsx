import * as React from "react";
import { cx } from "emotion";
import Box, { BoxProps } from "./Box";
import {
  BoxSides,
  SpaceSize
} from "../../../shared/styles/styleUtils/modifiers/modifierUtils";
import { padding } from "../../../shared/styles/styleUtils";

interface SpacingBoxProps extends BoxProps {
  /**
   * Which side (or sides) the spacing should be on
   */
  side?: BoxSides;
  /**
   * The size of the space to apply. It can set 1 spacing size for all breakpoints, or it can be used to set different spacing values at different viewport width breakpoints
   */
  spacingSize?: SpaceSize;
}

const SpacingBox = (props: SpacingBoxProps) => {
  const { side, spacingSize, className, ...other } = props;

  return (
    <Box
      className={cx(className, padding(side, spacingSize))}
      dataCy="spacingBox"
      {...other}
    />
  );
};

SpacingBox.defaultProps = {
  side: "all",
  spacingSize: "m",
  tag: "div"
};

export default SpacingBox;
