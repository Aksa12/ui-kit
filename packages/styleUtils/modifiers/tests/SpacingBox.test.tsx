import React from "react";
import { shallow } from "enzyme";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";
import toJson from "enzyme-to-json";

import SpacingBox from "../components/SpacingBox";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("SpacingBox", () => {
  it("renders default", () => {
    const component = shallow(<SpacingBox tag="div">content</SpacingBox>);

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders with all props", () => {
    const component = shallow(
      <SpacingBox tag="div" side="top" spacingSize="l">
        content
      </SpacingBox>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders with responsive spacingSize", () => {
    const component = shallow(
      <SpacingBox
        tag="div"
        side="top"
        spacingSize={{ default: "s", medium: "m" }}
      >
        content
      </SpacingBox>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
