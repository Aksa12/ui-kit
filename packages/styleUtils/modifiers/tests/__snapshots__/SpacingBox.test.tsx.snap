// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpacingBox renders default 1`] = `
.emotion-0 {
  padding: 16px;
}

<Box
  bgImageOptions={
    Object {
      "position": undefined,
      "repeat": undefined,
      "size": undefined,
    }
  }
  className="emotion-0"
  tag="div"
>
  content
</Box>
`;

exports[`SpacingBox renders with all props 1`] = `
.emotion-0 {
  padding-top: 24px;
}

<Box
  bgImageOptions={
    Object {
      "position": undefined,
      "repeat": undefined,
      "size": undefined,
    }
  }
  className="emotion-0"
  tag="div"
>
  content
</Box>
`;

exports[`SpacingBox renders with responsive spacingSize 1`] = `
@media (min-width:0) {
  .emotion-0 {
    padding-top: 12px;
  }
}

@media (min-width:900px) {
  .emotion-0 {
    padding-top: 16px;
  }
}

<Box
  bgImageOptions={
    Object {
      "position": undefined,
      "repeat": undefined,
      "size": undefined,
    }
  }
  className="emotion-0"
  tag="div"
>
  content
</Box>
`;
